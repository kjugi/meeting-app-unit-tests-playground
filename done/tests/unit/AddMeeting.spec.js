import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import axios from 'axios'
// Check for more: https://github.com/ctimmerm/axios-mock-adapter#axios-mock-adapter
import MockArapter from 'axios-mock-adapter'
// Check for more: https://github.com/kentor/flush-promises
// or if you don't want to add external lib: https://github.com/kentor/flush-promises/blob/master/index.js
import flushPromises from 'flush-promises'
import AddMeeting from '@/views/AddMeeting.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

// Check for more: https://jestjs.io/docs/en/timer-mocks
jest.useFakeTimers()

describe('AddMeeting page', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      doReservation: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })

    const mock = new MockArapter(axios)

    mock.onGet('http://localhost:5679/users').reply(200, [
      {
        text: 'test',
        value: 'example@o2.pl'
      },
      {
        text: 'Example Person1',
        value: 'meeting1@gmail.com'
      },
      {
        text: 'Example Person2',
        value: 'meeting2@gmail.com'
      }
    ])
    mock.onPost('http://localhost:5679/add').reply(200, true)
  })

  it('default form is rendered', () => {
    const wrapper = mount(AddMeeting)

    expect(wrapper.classes()).toStrictEqual([
      'add-meeting',
      'add-meeting--loading'
    ])
    expect(wrapper.find('#email').exists()).toBe(true)
    expect(wrapper.find('#predefined').exists()).toBe(true)
    expect(wrapper.find({ ref: 'meetingDate' }).exists()).toBe(true)
    expect(wrapper.find('.add-meeting__button').exists()).toBe(true)
    expect(wrapper.find('.add-meeting__error').exists()).toBe(true)
  })

  it('has loading class when form is blocked', async () => {
    const wrapper = mount(AddMeeting)

    expect(wrapper.vm.isFormBlocked).toBe(true)
    expect(wrapper.classes('add-meeting--loading')).toBe(true)

    await flushPromises()
    wrapper.setData({ isFormBlocked: false })

    expect(wrapper.vm.isFormBlocked).toBe(false)
    expect(wrapper.classes('add-meeting--loading')).toBe(false)
  })

  it('predefined addreses are available', async () => {
    const wrapper = mount(AddMeeting)

    expect(wrapper.vm.options).toBeNull()

    await flushPromises()

    expect(wrapper.vm.isMessageShowed).toBe(false)
    expect(wrapper.vm.messageTitle).toBe('')
    expect(wrapper.vm.options.length).toBe(3)
    expect(wrapper.findAll('#email option').length).toBe(4)
  })

  it('predefined select change value to selected option', async () => {
    const wrapper = mount(AddMeeting)
    await flushPromises()

    expect(wrapper.vm.selectedPerson).toBe('')

    wrapper.findAll('#email option').at(2).setSelected()

    expect(wrapper.vm.selectedPerson).toBe('Example Person1')
  })

  it('predefined checkbox toggle fields and clear values', async () => {
    const wrapper = mount(AddMeeting)
    await flushPromises()

    expect(wrapper.vm.predefined).toBe(true)

    await wrapper.find('#predefined').setChecked(false)

    expect(wrapper.vm.predefined).toBe(false)

    wrapper.find('#email').setValue('example value')

    expect(wrapper.vm.email).toBe('example value')

    await wrapper.find('#predefined').setChecked(true)

    expect(wrapper.vm.email).toBe('')
    expect(wrapper.find('#email').exists()).toBe(true)
  })

  it('can set custom meeting day in form', async () => {
    const wrapper = mount(AddMeeting)
    const todayDateFormat = new Date().toJSON().slice(0, 10)
    await flushPromises()

    expect(wrapper.vm.meetingDate).toBe('')

    wrapper.find('#meetingDate').setValue(todayDateFormat)

    expect(wrapper.find('#meetingDate').element.value).toBe(todayDateFormat)
    expect(wrapper.vm.meetingDate).toBe(todayDateFormat)
  })

  it('min attribute in input date has today date', async () => {
    const wrapper = mount(AddMeeting)
    const todayDateFormat = new Date().toJSON().slice(0, 10)
    await flushPromises()

    expect(wrapper.find('#meetingDate').attributes().min).toBe(todayDateFormat)
  })

  it('error is showed when form is invalid', async () => {
    const wrapper = mount(AddMeeting)

    expect(wrapper.vm.isFormValid).toBe(false)
    expect(wrapper.find('.add-meeting__error').exists()).toBe(true)

    await wrapper.setData({
      email: 'test',
      meetingDate: '2020-12-12'
    })

    expect(wrapper.vm.isFormValid).toBe(true)
    expect(wrapper.find('.add-meeting__error').exists()).toBe(false)
  })

  it('button can be disable or enabled', async () => {
    const wrapper = mount(AddMeeting)
    await flushPromises()

    expect(wrapper.vm.isFormValid).toBe(false)
    expect(wrapper.vm.isFormBlocked).toBe(false)
    expect(wrapper.find('.add-meeting__button').attributes().disabled).toBe('disabled')

    await wrapper.setData({
      email: 'test',
      meetingDate: '2020-12-12'
    })

    expect(wrapper.vm.isFormValid).toBe(true)
    expect(wrapper.find('.add-meeting__button').attributes().disabled).toBeUndefined()
  })

  it('can add meeting to store', async () => {
    const wrapper = mount(AddMeeting, { store, localVue })
    await flushPromises()
    await wrapper.setData({
      email: 'test',
      meetingDate: '2020-12-12'
    })

    wrapper.find('.add-meeting__button').trigger('click')

    expect(wrapper.vm.isFormBlocked).toBe(true)

    await flushPromises()

    expect(actions.doReservation).toHaveBeenCalled()
    expect(wrapper.vm.isFormBlocked).toBe(false)
    expect(wrapper.vm.email).toBe('')
    expect(wrapper.vm.meetingDate).toBe('')
  })

  it('message is showed after button click and hide on @hideMessage event', async () => {
    const wrapper = mount(AddMeeting, { store, localVue })
    await flushPromises()
    await wrapper.setData({
      email: 'test',
      meetingDate: '2020-12-12'
    })
    wrapper.find('.add-meeting__button').trigger('click')
    await flushPromises()

    expect(wrapper.vm.isMessageShowed).toBe(true)
    expect(wrapper.vm.messageClass).toBe('message--success')
    expect(wrapper.vm.messageTitle).toBe('Successfully added a new meeting')
    expect(wrapper.find('.message--success').isVisible()).toBe(true)

    jest.runAllTimers()
    await flushPromises()

    expect(wrapper.vm.isMessageShowed).toBe(false)
    expect(wrapper.find('.message--success').isVisible()).toBe(false)
  })
})
