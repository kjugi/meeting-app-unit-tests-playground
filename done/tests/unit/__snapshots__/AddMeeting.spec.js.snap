// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/AddMeeting.vue button can be disable or enabled 1`] = `
<div class="add-meeting">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
          <option data-testid="predefinedEmailOption">
            test
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person1
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person2
          </option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue button can be disable or enabled 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -21,16 +21,14 @@
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>
      <div class=\\"add-meeting__wrapper\\"><label for=\\"meetingDate\\">
          Meeting date
        </label> <input id=\\"meetingDate\\" data-testid=\\"meetingDateInput\\" type=\\"date\\" min=\\"2000-01-01\\" class=\\"add-meeting__input\\"></div>
-     <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" disabled=\\"disabled\\" class=\\"add-meeting__button\\">
+     <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" class=\\"add-meeting__button\\">
          Save meeting
        </button>
-       <p data-testid=\\"generalErrorMessage\\" class=\\"add-meeting__error\\">
-         You've got a problem in form
-       </p>
+       <!---->
      </div>
    </div>
    <div class=\\"message\\" style=\\"display: none;\\">
      <h2>
"
`;

exports[`views/AddMeeting.vue can add meeting to store 1`] = `
<div class="add-meeting add-meeting--loading">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue can add meeting to store 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,27 +1,23 @@
- <div class=\\"add-meeting add-meeting--loading\\">
+ <div class=\\"add-meeting\\">
    <div class=\\"add-meeting__container\\">
      <h1>Add meeting</h1>
      <div class=\\"add-meeting__wrapper\\">
        <div class=\\"add-meeting__field\\"><label for=\\"email\\">
            Choose person to meet
-         </label> <select id=\\"email\\" data-testid=\\"emailInput\\" class=\\"add-meeting__input\\">
-           <option value=\\"\\" selected=\\"selected\\" data-testid=\\"predefinedEmailOption\\"></option>
-         </select></div>
+         </label> <input id=\\"email\\" data-testid=\\"emailInput\\" type=\\"text\\" placeholder=\\"Email address\\" class=\\"add-meeting__input\\"></div>
        <div class=\\"checkbox add-meeting__field\\"><label for=\\"predefined\\">
            Predefined?
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>
      <div class=\\"add-meeting__wrapper\\"><label for=\\"meetingDate\\">
          Meeting date
        </label> <input id=\\"meetingDate\\" data-testid=\\"meetingDateInput\\" type=\\"date\\" min=\\"2000-01-01\\" class=\\"add-meeting__input\\"></div>
-     <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" disabled=\\"disabled\\" class=\\"add-meeting__button\\">
+     <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" class=\\"add-meeting__button\\">
          Save meeting
        </button>
-       <p data-testid=\\"generalErrorMessage\\" class=\\"add-meeting__error\\">
-         You've got a problem in form
-       </p>
+       <!---->
      </div>
    </div>
    <div class=\\"message\\" style=\\"display: none;\\">
      <h2>
"
`;

exports[`views/AddMeeting.vue can add meeting to store 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -10,17 +10,19 @@
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>
      <div class=\\"add-meeting__wrapper\\"><label for=\\"meetingDate\\">
          Meeting date
        </label> <input id=\\"meetingDate\\" data-testid=\\"meetingDateInput\\" type=\\"date\\" min=\\"2000-01-01\\" class=\\"add-meeting__input\\"></div>
-     <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" class=\\"add-meeting__button\\">
+     <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" class=\\"add-meeting__button\\" disabled=\\"disabled\\">
          Save meeting
        </button>
-       <!---->
+       <p data-testid=\\"generalErrorMessage\\" class=\\"add-meeting__error\\">
+         You've got a problem in form
+       </p>
      </div>
    </div>
-   <div class=\\"message\\" style=\\"display: none;\\">
+   <div class=\\"message message--success\\" style=\\"\\">
      <h2>
-
+       Successfully added a new meeting
      </h2>
    </div>
  </div>"
`;

exports[`views/AddMeeting.vue can set custom meeting day in form 1`] = `
<div class="add-meeting">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
          <option data-testid="predefinedEmailOption">
            test
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person1
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person2
          </option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue can set custom meeting day in form 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`views/AddMeeting.vue default form is rendered 1`] = `
<div class="add-meeting add-meeting--loading">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue error is showed when form is invalid 1`] = `
<div class="add-meeting add-meeting--loading">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue error is showed when form is invalid 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -15,13 +15,11 @@
          Meeting date
        </label> <input id=\\"meetingDate\\" data-testid=\\"meetingDateInput\\" type=\\"date\\" min=\\"2000-01-01\\" class=\\"add-meeting__input\\"></div>
      <div class=\\"add-meeting__action-wrapper\\"><button data-testid=\\"addMeetingButton\\" disabled=\\"disabled\\" class=\\"add-meeting__button\\">
          Save meeting
        </button>
-       <p data-testid=\\"generalErrorMessage\\" class=\\"add-meeting__error\\">
-         You've got a problem in form
-       </p>
+       <!---->
      </div>
    </div>
    <div class=\\"message\\" style=\\"display: none;\\">
      <h2>
"
`;

exports[`views/AddMeeting.vue has loading class when form is blocked 1`] = `
<div class="add-meeting add-meeting--loading">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue has loading class when form is blocked 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,13 +1,22 @@
- <div class=\\"add-meeting add-meeting--loading\\">
+ <div class=\\"add-meeting\\">
    <div class=\\"add-meeting__container\\">
      <h1>Add meeting</h1>
      <div class=\\"add-meeting__wrapper\\">
        <div class=\\"add-meeting__field\\"><label for=\\"email\\">
            Choose person to meet
          </label> <select id=\\"email\\" data-testid=\\"emailInput\\" class=\\"add-meeting__input\\">
            <option value=\\"\\" selected=\\"selected\\" data-testid=\\"predefinedEmailOption\\"></option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             test
+           </option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             Example Person1
+           </option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             Example Person2
+           </option>
          </select></div>
        <div class=\\"checkbox add-meeting__field\\"><label for=\\"predefined\\">
            Predefined?
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>"
`;

exports[`views/AddMeeting.vue predefined addreses are available 1`] = `
<div class="add-meeting add-meeting--loading">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue predefined addreses are available 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,13 +1,22 @@
- <div class=\\"add-meeting add-meeting--loading\\">
+ <div class=\\"add-meeting\\">
    <div class=\\"add-meeting__container\\">
      <h1>Add meeting</h1>
      <div class=\\"add-meeting__wrapper\\">
        <div class=\\"add-meeting__field\\"><label for=\\"email\\">
            Choose person to meet
          </label> <select id=\\"email\\" data-testid=\\"emailInput\\" class=\\"add-meeting__input\\">
            <option value=\\"\\" selected=\\"selected\\" data-testid=\\"predefinedEmailOption\\"></option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             test
+           </option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             Example Person1
+           </option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             Example Person2
+           </option>
          </select></div>
        <div class=\\"checkbox add-meeting__field\\"><label for=\\"predefined\\">
            Predefined?
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>"
`;

exports[`views/AddMeeting.vue predefined checkbox toggle fields and clear values 1`] = `
<div class="add-meeting">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
          <option data-testid="predefinedEmailOption">
            test
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person1
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person2
          </option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue predefined checkbox toggle fields and clear values 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,22 +2,11 @@
    <div class=\\"add-meeting__container\\">
      <h1>Add meeting</h1>
      <div class=\\"add-meeting__wrapper\\">
        <div class=\\"add-meeting__field\\"><label for=\\"email\\">
            Choose person to meet
-         </label> <select id=\\"email\\" data-testid=\\"emailInput\\" class=\\"add-meeting__input\\">
-           <option value=\\"\\" selected=\\"selected\\" data-testid=\\"predefinedEmailOption\\"></option>
-           <option data-testid=\\"predefinedEmailOption\\">
-             test
-           </option>
-           <option data-testid=\\"predefinedEmailOption\\">
-             Example Person1
-           </option>
-           <option data-testid=\\"predefinedEmailOption\\">
-             Example Person2
-           </option>
-         </select></div>
+         </label> <input id=\\"email\\" data-testid=\\"emailInput\\" type=\\"text\\" placeholder=\\"Email address\\" class=\\"add-meeting__input\\"></div>
        <div class=\\"checkbox add-meeting__field\\"><label for=\\"predefined\\">
            Predefined?
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>
      <div class=\\"add-meeting__wrapper\\"><label for=\\"meetingDate\\">"
`;

exports[`views/AddMeeting.vue predefined checkbox toggle fields and clear values 3`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`views/AddMeeting.vue predefined checkbox toggle fields and clear values 4`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,11 +2,22 @@
    <div class=\\"add-meeting__container\\">
      <h1>Add meeting</h1>
      <div class=\\"add-meeting__wrapper\\">
        <div class=\\"add-meeting__field\\"><label for=\\"email\\">
            Choose person to meet
-         </label> <input id=\\"email\\" data-testid=\\"emailInput\\" type=\\"text\\" placeholder=\\"Email address\\" class=\\"add-meeting__input\\"></div>
+         </label> <select id=\\"email\\" data-testid=\\"emailInput\\" class=\\"add-meeting__input\\">
+           <option value=\\"\\" selected=\\"selected\\" data-testid=\\"predefinedEmailOption\\"></option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             test
+           </option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             Example Person1
+           </option>
+           <option data-testid=\\"predefinedEmailOption\\">
+             Example Person2
+           </option>
+         </select></div>
        <div class=\\"checkbox add-meeting__field\\"><label for=\\"predefined\\">
            Predefined?
          </label> <input id=\\"predefined\\" type=\\"checkbox\\" data-testid=\\"predefinedCheckbox\\"></div>
      </div>
      <div class=\\"add-meeting__wrapper\\"><label for=\\"meetingDate\\">"
`;

exports[`views/AddMeeting.vue predefined select change value to selected option 1`] = `
<div class="add-meeting">
  <div class="add-meeting__container">
    <h1>Add meeting</h1>
    <div class="add-meeting__wrapper">
      <div class="add-meeting__field"><label for="email">
          Choose person to meet
        </label> <select id="email" data-testid="emailInput" class="add-meeting__input">
          <option value="" selected="selected" data-testid="predefinedEmailOption"></option>
          <option data-testid="predefinedEmailOption">
            test
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person1
          </option>
          <option data-testid="predefinedEmailOption">
            Example Person2
          </option>
        </select></div>
      <div class="checkbox add-meeting__field"><label for="predefined">
          Predefined?
        </label> <input id="predefined" type="checkbox" data-testid="predefinedCheckbox"></div>
    </div>
    <div class="add-meeting__wrapper"><label for="meetingDate">
        Meeting date
      </label> <input id="meetingDate" data-testid="meetingDateInput" type="date" min="2000-01-01" class="add-meeting__input"></div>
    <div class="add-meeting__action-wrapper"><button data-testid="addMeetingButton" disabled="disabled" class="add-meeting__button">
        Save meeting
      </button>
      <p data-testid="generalErrorMessage" class="add-meeting__error">
        You've got a problem in form
      </p>
    </div>
  </div>
  <div class="message" style="display: none;">
    <h2>

    </h2>
  </div>
</div>
`;

exports[`views/AddMeeting.vue predefined select change value to selected option 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
